src/cdomains/apron/linearTwoVarEqualityDomain.apron.ml

```diff
@@ -0,0 +1,952 @@
+        | _ -> raise NotLinearExpr
+      else raise NotLinearExpr in
+    let mpqf_to_Z x = 
+      if not(Z.equal (Mpqf.get_den x) Z.one) then raise NotIntegerOffset
+      else Mpqf.get_num x in
+    let rec convert_texpr texp =
+      begin match texp with
+        (*If x is a constant, replace it with its const. val. immediately*)
+        | Cst x -> let of_union union =
+                     let open Coeff in
+                     match union with
+                     | Interval _ -> failwith "Not a constant"
+                     | Scalar x -> begin match x with
+                         | Float x -> raise NotIntegerOffset
+                         | Mpqf x -> [(mpqf_to_Z x, None)]
+                         | Mpfrf x -> raise NotIntegerOffset end in of_union x
+        | Var x -> 
+          let var_dim = Environment.dim_of_var t.env x in 
+          begin match get_variable_value_if_it_is_a_constant t var_dim with
+            | None -> [(Z.one, Some var_dim)]
+            | Some constant -> [(constant, None)]
+          end
+        | Unop (u, e, _, _) ->
+          begin match u with
+            | Neg -> negate (convert_texpr e)
+            | Cast -> convert_texpr e (*Ignore since casts in apron are used for floating point nums and rounding in contrast to CIL casts*)
+            | Sqrt -> raise NotLinearExpr end
+        | Binop (b, e1, e2, _, _) ->
+          begin match b with
+            | Add -> List.concat [convert_texpr e1; convert_texpr e2]
+            | Sub -> List.concat [convert_texpr e1; negate (convert_texpr e2)]
+            | Mul -> multiply (convert_texpr e1) (convert_texpr e2)
+            | _ -> raise NotLinearExpr end
+      end
+    in convert_texpr texp 
+
+  let get_coeff (t: t) texp =
+    (*Parses a Texpr to obtain a (variable, offset) pair to repr. a sum of a variable and an offset.
+      Returns None if the expression is not a sum between a variable (without coefficient) and a constant. 
+    *)
+    let exception Not2VarExpr in
+    let sum_coefficients summands_list =
+      List.fold_left (fun (var, current_var_offset, curr_offset) (next_coeff, next_var) -> 
+          begin match next_var with
+            | None -> (* this element represents a constant offset *)
+              (var, current_var_offset, Z.(curr_offset + next_coeff))
+            | Some same_var -> (* this element represents a variable with a coefficient 
+                                  -> it must be always the same variable, else it's not a two-variable equality*)
+              begin if Option.is_none var || Some same_var = var then
+                  (Some same_var, Z.(current_var_offset + next_coeff), curr_offset) 
+                else raise Not2VarExpr end 
+          end)
+        (None, Z.zero, Z.zero) summands_list
+    in
+    match get_coeff_vec t texp with
+    | exception _ -> None
+    | summands_list -> 
+      match sum_coefficients summands_list with
+      | exception _ -> None 
```

Comment:
Can one avoid these nested `match`-statments? E.g. by using a `try` or directly calling `sum_coefficients` in the same statement?